### PR TITLE
fix: auto fields update include tests. before code review

### PR DESCRIPTION
**Pull request by itaykl**

### CODE REVIEW

File: src/subscribers/common-model-event-subscriber.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        if (event.entity) {
+            event.connection.logger.log('log', 'prePersist began', event.queryRunner);
+            await runAndMeasureTime(async () => {
+                event.entity.setDataVersion(this.getExtensions(event.manager).globalDataVersion);
+                const now = new Date(
+                    new Date().toLocaleString('en-US', { timeZone: 'Asia/Jerusalem' }),
+                );
+                event.entity.setCreationTime(now);
+                event.entity.setLastUpdateTime(now);
+                event.entity.setCreatedBy(this.getUpnOrRequestingSystemIdFromHeader(event.manager));
```
Comment (chenshoo) [inline src/subscribers/common-model-event-subscriber.ts]: this.getUpnOrRequestingSystemIdFromHeader(event.manager) put in var

File: src/subscribers/common-model-event-subscriber.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        }
+    }
+
+    public async beforeUpdate(event: UpdateEvent<CommonModel>) {
+        if (event.entity) {
+            event.connection.logger.log('log', 'preUpdate began');
+            await runAndMeasureTime(async () => {
+                event.entity.setDataVersion(this.getExtensions(event.manager).globalDataVersion);
+                const now = new Date(
+                    new Date().toLocaleString('en-US', { timeZone: 'Asia/Jerusalem' }),
```
Comment (chenshoo) [inline src/subscribers/common-model-event-subscriber.ts]: also here new Date()

File: src/subscribers/common-model-event-subscriber.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            throw new Error('reality id of entity is different from header');
+        }
+    }
+
+    private getUpnOrRequestingSystemIdFromHeader(manager: EntityManager): string | undefined {
+        const headers: PolarisRequestHeaders =
+            manager.queryRunner &&
+            manager.queryRunner.data &&
+            (manager.queryRunner.data.requestHeaders || {});
+        return headers.upn ? headers.upn : headers.requestingSystemId;
```
Comment (chenshoo) [inline src/subscribers/common-model-event-subscriber.ts]: headers.upn || headers.requestingSystemId

File: test/unit-tests/subscribers/common-model-event-subscriber.test.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+let bookEntity: Book;
+let commonModelEventSubscriber: CommonModelEventSubscriber;
+
+describe('common model event subscriber tests', () => {
+    beforeEach(async () => {
+        bookEntity = new Book();
+        commonModelEventSubscriber = new CommonModelEventSubscriber();
+    });
+    it('beforeInsert - all the common model fields defined as needed', async () => {
```
Comment (chenshoo) [inline test/unit-tests/subscribers/common-model-event-subscriber.test.ts]: describe before insert

File: test/unit-tests/subscribers/common-model-event-subscriber.test.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        expect(bookEntity.getDataVersion()).toEqual(1);
+        expect(bookEntity.getRealityId()).toEqual(0);
+    });
+    it('beforeInsert - check that the logger called as needed', async () => {
+        const connection = setConnection(1);
+        const event = new DummyInsertEvent(connection, bookEntity);
+        await commonModelEventSubscriber.beforeInsert(event);
+        expect(connection.logger.log).toBeCalledTimes(2);
+    });
+    it('beforeUpdate - all the common model fields defined as needed', async () => {
```
Comment (chenshoo) [inline test/unit-tests/subscribers/common-model-event-subscriber.test.ts]: describe before update

File: src/models/common-model.ts [hunk 6]
```diff
@@ -61,22 +61,22 @@ export abstract class CommonModel {
...
-    public setDataVersion(dataVersion: string) {
+    public setDataVersion(dataVersion: number) {
         this.dataVersion = dataVersion;
     }
     public setRealityId(realityId: number) {
         this.realityId = realityId;
     }
 
-    public setCreatedBy(createdBy: string) {
+    public setCreatedBy(createdBy: string | undefined) {
```
Comment (DoctorVoid) [inline src/models/common-model.ts]: createdBy?: string

do this for all the next function as well

**Pull request merged**